File: A_00/primes.js
function is_prime(n) {
    if (n <= 1) { return false; }
    for (let i = 2; i < Math.floor(Math.sqrt(n))+1; i++) {
        if (n % i == 0) { return false; }
    }
    return true;
}

function primes(a, b) {
    prime_list = [];
    for(let i = a; i < (b + 1); i++) {
        if(is_prime(i)) { prime_list.push(i); }
    }
    return prime_list;
}

function prime_partitions(n, k = 1, lst = []) {
    if(n==0) {
        print_primes(lst)
    } else if(n > k) {
        primes(k + 1, n).forEach((i)=> {
            prime_partitions(n - i, i, lst.concat(i));
        })
    }
}

function print_primes(lst) {
    let str = ""
    for (let i = 0; i < lst.length - 1; i++) {
        str += lst[i] + " + ";
    }
    str += lst[lst.length-1];
    console.log(str)
}

//Runs the Program
const readline = require('readline')
var rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

function ask_user() {
    rl.question("Enter a number (non-number to quit): ", (answer) => {
        answer = parseInt(answer)
        if(isNaN(answer)) {
            console.log(`\n\nBye!`);
            process.exit();
        } else {
            prime_partitions(answer);
            ask_user();
        }
    });
}

ask_user();

File: A_00/primes.go
package main

import (
    "fmt"
    "math"
    "strings"
    "strconv"
)

func main() {
    fmt.Printf("Enter a number (non-number to quit): ")

    var num int
    // read in user input
    num_read, err := fmt.Scanf("%d", &num)

    // fmt.Printf("num read: %d", num_read)

    if(num_read < 1 && err != nil) {
        fmt.Println("\n\nBye!")
        
    } else {
        var lst = []int{}
        prime_partitions(num, 1, lst)
    }

}


/**
 * Function determines if a number is a prime
 **/
func is_prime(n int) (bool) {
    if n <= 1 {
        return false
    }
    for i:=2; i <  int(math.Sqrt(float64(n))) + 1; i++ {
        if n % i == 0 {
            return false
        }
    }
    return true
}

func primes(a int, b int) ([]int) {
    var primes_list = []int{}
    for i:=a; i < b + 1; i++ {
        if(is_prime(i)) {
            primes_list = append(primes_list, i)
        }
    }

    return primes_list
}

func prime_partitions(n int, k int, lst []int) {
    if(n == 0){
        print_primes(lst)
    } else if(n > k) {
        for _, p := range primes(k + 1, n) {
            // fmt.Printf("%d", i)
            prime_partitions(n - p, p, append(lst, p))
        }
    }
}

func print_primes(lst []int) {
    print_list := []string{}
    for i:=0; i < len(lst); i++ {
        val := strconv.Itoa(lst[i])
        print_list = append(print_list, val)
    }
    fmt.Println(strings.Join(print_list, " + "))
}
